src/audio/copier/copier.c

```diff
@@ -740,6 +740,29 @@ static struct comp_dev *copier_new(const struct comp_driver *drv,
 	}
 
 	dev->pipeline = ipc_pipe->pipeline;
+	/* Calculation of the period in which the pipeline should be scheduled based on
+	 * the input buffer size.
+	 */
+	if (copier->gtw_cfg.dma_buffer_size > 0) {
+		/* Size of the single sample in bytes. */
+		uint32_t sample_size = copier->base.audio_fmt.channels_count *
+				       (copier->base.audio_fmt.depth / 8);
+		/* Size of the one millisecond of audio data in bytes. */
+		uint32_t one_ms = (sample_size * copier->base.audio_fmt.sampling_frequency) / 1000;
+		/* The number of milliseconds of data that will fit into the buffer. */
+		uint32_t ms_count = copier->gtw_cfg.dma_buffer_size / one_ms;
+
+
+		__ASSERT_NO_MSG(ms_count > 0);
+		if (ms_count == 0)
+			ms_count = 1;
+
+		/* By default, the period was set to one millisecond, so now just multiply it
+		 * to get a new value.
+		 */
+		dev->pipeline->period = dev->pipeline->period * ms_count;
```

Comment:
What if there are two copiers, each one attached to a gateway? The second one overrides the period no matter what are the constraints of the first one?